backend/src/utils/tests.ts

```diff
@@ -0,0 +1,18 @@
+import { pool } from '../db/pool';
+import { waitForDatabase } from '../db/util';
+
+const timeout = 60;
+
+jest.setTimeout(timeout * 1000);
+
+beforeAll(async () => {
+    await waitForDatabase(pool, timeout);
+});
+
+beforeEach(async () => {
+    await pool.query('BEGIN');
```

Comment:
beforeEach `BEGIN` will fail if any further `BEGIN` calls are made. A more stable way is to run a db reset + migrations on beforeEach.

Reply:
Resolved by https://github.com/bytecraftoy/aalto-2022/pull/90/commits/0255fed71dedeffdb110237c80c38408fbc84df9